File: src/main/java/cn/cloudself/model/AppSliderEntity.java
package cn.cloudself.model;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *          <p>
 *          change logs:
 *          2017/3/28 HerbLuo 首次创建
 */
@Entity
@Table(name = "app_slider", schema = "shop")
public class AppSliderEntity {
    private int id;
    private byte enabled;
    private String imgSrc;
    private Integer itemId;
    private String link;
    private Integer category;
    private Integer shopId;

    @Id
    @Column(name = "id", nullable = false)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "enabled", nullable = false)
    public byte getEnabled() {
        return enabled;
    }

    public void setEnabled(byte enabled) {
        this.enabled = enabled;
    }

    @Basic
    @Column(name = "imgSrc", nullable = false, length = 128)
    public String getImgSrc() {
        return imgSrc;
    }

    public void setImgSrc(String imgSrc) {
        this.imgSrc = imgSrc;
    }

    @Basic
    @Column(name = "itemId", nullable = true)
    public Integer getItemId() {
        return itemId;
    }

    public void setItemId(Integer itemId) {
        this.itemId = itemId;
    }

    @Basic
    @Column(name = "link", nullable = true, length = 255)
    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    @Basic
    @Column(name = "category", nullable = true)
    public Integer getCategory() {
        return category;
    }

    public void setCategory(Integer category) {
        this.category = category;
    }

    @Basic
    @Column(name = "shopId", nullable = true)
    public Integer getShopId() {
        return shopId;
    }

    public void setShopId(Integer shopId) {
        this.shopId = shopId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AppSliderEntity that = (AppSliderEntity) o;

        if (id != that.id) return false;
        if (enabled != that.enabled) return false;
        if (imgSrc != null ? !imgSrc.equals(that.imgSrc) : that.imgSrc != null) return false;
        if (itemId != null ? !itemId.equals(that.itemId) : that.itemId != null) return false;
        if (link != null ? !link.equals(that.link) : that.link != null) return false;
        if (category != null ? !category.equals(that.category) : that.category != null) return false;
        if (shopId != null ? !shopId.equals(that.shopId) : that.shopId != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (int) enabled;
        result = 31 * result + (imgSrc != null ? imgSrc.hashCode() : 0);
        result = 31 * result + (itemId != null ? itemId.hashCode() : 0);
        result = 31 * result + (link != null ? link.hashCode() : 0);
        result = 31 * result + (category != null ? category.hashCode() : 0);
        result = 31 * result + (shopId != null ? shopId.hashCode() : 0);
        return result;
    }
}

File: src/main/java/cn/cloudself/model/AppSliderEntity.kt
package cn.cloudself.model

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *
 *
 * change logs:
 * 2017/3/28 HerbLuo 首次创建
 */
@Entity
@Table(name = "app_slider", schema = "shop")
data class AppSliderEntity(
        @get:Id
        @get:Column(name = "id", nullable = false)
        var id: Int = 0,
        @get:Basic
        @get:Column(name = "enabled", nullable = false)
        var enabled: Byte = 0,
        @get:Basic
        @get:Column(name = "imgSrc", nullable = false, length = 128)
        var imgSrc: String? = null,
        @get:Basic
        @get:Column(name = "itemId", nullable = true)
        var itemId: Int? = null,
        @get:Basic
        @get:Column(name = "link", nullable = true, length = 255)
        var link: String? = null,
        @get:Basic
        @get:Column(name = "category", nullable = true)
        var category: Int? = null,
        @get:Basic
        @get:Column(name = "shopId", nullable = true)
        var shopId: Int? = null
)
